core/src/metadata/metadata_type.rs

```diff
@@ -70,68 +76,6 @@ impl TryFrom<frame_metadata::RuntimeMetadataPrefixed> for Metadata {
 
 impl codec::Decode for Metadata {
     fn decode<I: codec::Input>(input: &mut I) -> Result<Self, codec::Error> {
-        subxt_metadata::Metadata::decode(input).map(Metadata::new)
-    }
-}
-
-/// Some extension methods on [`subxt_metadata::Metadata`] that return Errors instead of Options.
-pub trait MetadataExt {
```

Comment:
I didn't see a need for this offhand and just added the one missing function above on the concrete type instead.